### PR TITLE
Updated upgrade script that is informing user that Gitea service has to be running in order to upgrade it

### PR DESCRIPTION
**Pull request by roy9495**
Hello sir,
This pull request solves issue #23949
I have updated the upgrade script such that it will check if the Gitea service is running, if it's not running then it will exit the process and if it's running then the process will move further ahead.
Thank You.

### CODE REVIEW

File: contrib/upgrade.sh [hunk 1]
```diff
@@ -10,6 +10,17 @@
...
+
+
+# Check if gitea service is running
+if ! systemctl is-active gitea &> /dev/null; then
+  echo "Error: gitea is not running."
+  exit 1
+fi
+
+# Continue with rest of the script if gitea is running
+echo "Gitea is running. Continuing with rest of script..."
```
Comment (delvh) [inline contrib/upgrade.sh]: Unfortunately, it's not that easy.
As you can see a few lines below, there is no guarantee that a user has or uses `systemctl` as they can set themselves what they use.

**Review by delvh — CHANGES_REQUESTED** [review]

File: contrib/upgrade.sh (hunk 1 above)
Comment (wxiaoguang) [reply contrib/upgrade.sh]: Yup, the `service_start` command below is optional.

There should a configurable `service_is_active` command like `service_start` or `service_status`.

ps: could the exit code of `systemctl status` be used directly?

Comment (silverwind) [reply contrib/upgrade.sh]: `pidof gitea` maybe as a more standalone check? That should work everywhere except Windows.

Comment (roy9495) [reply contrib/upgrade.sh:22]: Sir can I do it like this?------
if ! pidof gitea &> /dev/null; then
  echo "Error: Gitea is not running."
  exit 1
fi

# Continue with rest of the script if Gitea is running
echo "Gitea is running. Continuing with rest of script..."

Comment (roy9495) [reply contrib/upgrade.sh]: Sir can I do it like this -----

Check if gitea service is running

if ! pidof gitea &> /dev/null; then
  echo "Error: Gitea is not running."
  exit 1
fi

 Continue with rest of the script if Gitea is running

echo "Gitea is running. Continuing with rest of script..."

Comment (silverwind) [reply contrib/upgrade.sh]: Reading the comment header of this file, I seems systemd can actually be assumed:

> this is an update script for gitea installed via the binary distribution
> from dl.gitea.io on linux as systemd service

So I guess the systemd method is fine, @delvh what do you think?

Comment (delvh) [reply contrib/upgrade.sh]: Hmm.
The rest of this file is independent of systemd, so I don't think it's a good idea to throw away that independence.
That will probably make exactly those people angry that don't use systemd…

Comment (silverwind) [reply contrib/upgrade.sh]: Ok, then `pidof gitea` it is.

Comment (techknowlogick) [reply contrib/upgrade.sh]: Yeah, I agree re: pidof, as even though systemctl is used within this file, you can still override those commands, especially if platform is openwrt then `/etc/init.d/gitea` will be used.

**Review by techknowlogick — APPROVED** [review]
Thanks so much!!

**Review by delvh — APPROVED** [review]

File: contrib/upgrade.sh [hunk 2]
```diff
@@ -10,6 +10,17 @@
 #   upgrade.sh 1.15.10
 #   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
 
+
+
```
Comment (silverwind) [inline contrib/upgrade.sh]:
```suggestion
```

**Review by silverwind — APPROVED** [review]

**Pull request merged**